### PR TITLE
Elementary syntactic structure

### PR DESCRIPTION
**Pull request by MrChico**

### CODE REVIEW

**Comment by leonardoalt** [comment]
Is the grammar included here for the low or high level Acts?

**Comment by MrChico** [comment]
The grammar supports both, with the production:
`Change. Update ::= Storage "=>" IExp ; ` for the high level syntax and
` OChange. Update ::= Storage "|->" Exp "=>" Exp ; ` for the verbose flavor

**Comment by MrChico** [comment]
Actually, I decided to take a more definite stance and have abandoned the old verbose syntax for now

**Comment by MrChico** [comment]
@leonardoalt this is ready to be merged. The main features included here are:
- Formal grammar for `behaviour` sections, as in `act.cf`.
- Some "concept art" specs in the `examples` directory
- Test files in `test` dir meant to keep up to date and added to as new grammar feature are introduced
- Some building infrastructure (`shell.nix`, `Makefile`).

File: README.md [hunk 1]
```diff
@@ -17,3 +17,100 @@ Starting Point
 The [Act](https://github.com/dapphub/klab/blob/master/acts.md) specification language already matches some of the desired features: It has a simple structure, simple logic, and supports properties written for EVM bytecode.
 
 It is currently being used as the starting point for this project, requiring few modifications to the structure and syntax such that it can also support high level properties.
+
+Infrastructure
+==============
+The grammar for the specification language is in the `src` repository. This is the front end parsing of the language. Given set of `act` behaviours (transitions), one can generate a set of proof obligations, expressed as a json object:
```
Comment (leonardoalt) [inline README.md]:
```suggestion
The grammar for the specification language is in the `src` repository. This is the front end parsing of the language. Given a set of `act` behaviours (transitions), one can generate a set of proof obligations, expressed as a JSON object:
```

File: README.md [hunk 2]
```diff
@@ -17,3 +17,100 @@ Starting Point
...
+   "0 <= v0",
+   "v0 < 2^256",
+   "0 <= v1",
+   "v0 < 2^256",
+   "v0 - amount <= v0",
+   "v0 - amount < 2^256",
+   "v1 + amount <= v0",
+   "v1 + amount < 2^256"
+ ],
+ "postcondition": []
```
Comment (leonardoalt) [inline README.md]: So in this format where each JSON object is a case, the postcondition would be repeated in each object, right?

File: shell.nix [hunk 3]
```diff
@@ -0,0 +1,5 @@
+with import <nixpkgs> {}; with haskellPackages;
+stdenv.mkDerivation {
+  name = "act";
+  buildInputs = [ BNFC alex happy ];
```
Comment (leonardoalt) [inline shell.nix]: What's `alex happy`?

File: README.md (hunk 2 above)
Comment (MrChico) [reply README.md]: Yes

File: shell.nix (hunk 3 above)
Comment (MrChico) [reply shell.nix]: Two haskell packages that `bnfc` invokes in order to generate the parsers.
Alex is a lexical analyser generator https://www.haskell.org/alex/
Happy is a parser generator https://www.haskell.org/happy/

**Pull request merged**